packages/maven/maven-plugin/src/main/kotlin/dev/nx/maven/targets/TestClassDiscovery.kt

```diff
@@ -129,14 +129,15 @@ class TestClassDiscovery() {
     }
 
     /**
-     * Extract public class name from Java file content using simple string matching
+     * Extract class name from Java file content using simple string matching
+     * Matches any class declaration, not just public ones (test classes can be package-private)
      */
-    private fun extractPublicClassName(content: String): String? {
+    private fun extractClassName(content: String): String? {
         val lines = content.lines()
         for (line in lines) {
             val trimmed = line.trim()
-            if (trimmed.contains("public class ")) {
-                // Simple extraction: find "public class ClassName"
+            if (trimmed.contains("class ")) {
+                // Simple extraction: find "class ClassName" (handles public, package-private, etc.)
```

Comment:
Test classes don't have to be public? At least the ones generated for Spring do not..